### PR TITLE
feat(secret): added support of Docker registry credentials

### LINKED ISSUE #5669: feat(secret): add support of Docker registry credentials
## Description
Add the following rule.
https://gist.github.com/yakirk/f0d5a4cb3893914f956336a9edd32cfe

### Discussed in https://github.com/aquasecurity/trivy/discussions/5664

### PR DESCRIPTION
**Pull request by sourav977**
Issue: https://github.com/aquasecurity/trivy/issues/5669

## Description
added support of Docker registry credentials

## Related issues
- Close #5669

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/fanal/secret/builtin-rules.go [hunk 1]
```diff
@@ -792,4 +793,13 @@ var builtinRules = []Rule{
...
 		Keywords:        []string{"typeform"},
 	},
+	{
+		ID:              "dockerconfig-secret",
+		Category:        CategoryDocker,
+		Title:           "Dockerconfig secret exposed",
+		Severity:        "HIGH",
+		Regex:           MustCompile(`(?i)(\.(dockerconfigjson|dockercfg):\s*\|*\s*(ey|ew)+)`),
+		SecretGroupName: "secret",
+		Keywords:        []string{"dockerconfig"},
```
Comment (DmitriyLewen) [inline pkg/fanal/secret/builtin-rules.go:803]: So as not to miss `dockercfg: ***` secrets.
```suggestion
		Keywords:        []string{"dockerc"},
```

**Review by DmitriyLewen — COMMENTED** [review]
Hello @sourav977 
Thanks for your work!

Left 2 comments. Take a look, please.

**Comment by sourav977** [comment]
@DmitriyLewen 
I worked on your review comments, please verify.

File: pkg/fanal/secret/builtin-rules.go [hunk 2]
```diff
@@ -792,4 +793,12 @@ var builtinRules = []Rule{
 		SecretGroupName: "secret",
 		Keywords:        []string{"typeform"},
 	},
+	{
+		ID:       "dockerconfig-secret",
+		Category: CategoryDocker,
+		Title:    "Dockerconfig secret exposed",
+		Severity: "HIGH",
+		Regex:    MustCompile(`(?i)(\.(dockerconfigjson|dockercfg):\s*\|*\s*(ey|ew)+)`),
```
Comment (DmitriyLewen) [inline pkg/fanal/secret/builtin-rules.go:801]: I think we can upgrade this regex.
If i understand correctly -  secret is base64 encoded string (` YAML files containing dockercfg or dockerconfigjson with base64-encoded secrets.`).
We can add secret group and mask only secret line:
```suggestion
		Regex:           MustCompile(`(?i)(\.(dockerconfigjson|dockercfg):\s*\|*\s*(?P<secret>(ey|ew)+[A-Za-z0-9\/\+=]+))`),
		SecretGroupName: "secret",
```

result before:
```bash
 asdf.txt:2
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   data:
   2 [   *********************JodHRwczovL2luZGV4L

```

result after:
```bash
 asdf.txt:2
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   data:
   2 [   .dockerconfigjson: *********************

```

**Review by DmitriyLewen — COMMENTED** [review]
Hm...
Looks like I forgot to save second comment about secret group, or GitHub deleted it.

Anyway, could you take a look at 1 more comment please?

File: pkg/fanal/secret/builtin-rules.go (hunk 2 above)
Comment (DmitriyLewen) [reply pkg/fanal/secret/builtin-rules.go]: We added `Secret group` (`?P<secret>`). 
Add `SecretGroupName: "secret",` now, please.

Comment (DmitriyLewen) [reply pkg/fanal/secret/builtin-rules.go]: Added in https://github.com/aquasecurity/trivy/pull/5720/commits/6022117f7336929dd8d97912bb8f6a7cf440a1ad

**Review by DmitriyLewen — APPROVED** [review]
Thanks for your work @sourav977 

@knqyf263 i approved this PR. Take a look, when you have time, please.

**Pull request merged**